tests/unit/test_navigation_table.py

```diff
@@ -0,0 +1,241 @@
+        pytest.param("|1|a|[a()|", True, id="third column closing square bracket missing"),
+        pytest.param("|1|a|[a])|", True, id="third column opening link bracket missing"),
+        pytest.param(r"|1|a|[a](\)|", True, id="third column link includes backslash"),
+        pytest.param("|1|a|[a](|", True, id="third column closing link bracket missing"),
+        pytest.param("|1|a|[a]()|", False, id="matches row"),
+    ],
+)
+def test__filter_line(line, expected_result):
+    """
+    arrange: given line and expected return value
+    act: when _filter_line is called with the line
+    assert: then the expected return value is returned.
+    """
+    returned_result = navigation_table._filter_line(line)
+
+    assert returned_result == expected_result
+
+
+@pytest.mark.parametrize(
+    "line, expected_result",
+    [
+        pytest.param("|1|a|[b]()|", (1, "a", ("b", None)), id="simple"),
+        pytest.param(" |1|a|[b]()|", (1, "a", ("b", None)), id="simple leading space"),
+        pytest.param("| 1|a|[b]()|", (1, "a", ("b", None)), id="simple space before first column"),
+        pytest.param("|1 |a|[b]()|", (1, "a", ("b", None)), id="simple space after first column"),
+        pytest.param(
+            "|1| a|[b]()|", (1, "a", ("b", None)), id="simple space before second column"
+        ),
+        pytest.param("|1|a |[b]()|", (1, "a", ("b", None)), id="simple space after second column"),
+        pytest.param("|1|a| [b]()|", (1, "a", ("b", None)), id="simple space before third column"),
+        pytest.param("|1|a|[b]() |", (1, "a", ("b", None)), id="simple space after third column"),
+        pytest.param("|1|a|[b]()| ", (1, "a", ("b", None)), id="simple trailing space"),
+        pytest.param("|12|a|[b]()|", (12, "a", ("b", None)), id="first column multiple digits"),
+        pytest.param(
+            "|1|az|[b]()|", (1, "az", ("b", None)), id="second column multiple characters"
+        ),
+        pytest.param("|1|A|[b]()|", (1, "A", ("b", None)), id="second column upper case"),
+        pytest.param("|1|2|[b]()|", (1, "2", ("b", None)), id="second column digit"),
+        pytest.param("|1|_|[b]()|", (1, "_", ("b", None)), id="second column underscore"),
+        pytest.param("|1|-|[b]()|", (1, "-", ("b", None)), id="second column dash"),
+        pytest.param(
+            "|1|a|[bz]()|", (1, "a", ("bz", None)), id="third column title multiple characters"
+        ),
+        pytest.param(
+            "|1|a|[B]()|", (1, "a", ("B", None)), id="third column title capital character"
+        ),
+        pytest.param("|1|a|[2]()|", (1, "a", ("2", None)), id="third column title digit"),
+        pytest.param("|1|a|[_]()|", (1, "a", ("_", None)), id="third column title underscore"),
+        pytest.param("|1|a|[-]()|", (1, "a", ("-", None)), id="third column title dash"),
+        pytest.param("|1|a|[ ]()|", (1, "a", (" ", None)), id="third column title space"),
```

Comment:
Is it really what we should expect ? I'd expect an empty string for this case no ?

Reply:
I went back and forth on this one a bit. Initially I was going to code it up so that the regular expression only captures the exact link title. This led to increasing complexity and I couldn't think of any value that the complexity brings since, if there is leading or trailing spaces, for example, in the title, the action would just end up cleaning that up anyway when the navigation table is generated and updated at the end. The reason this test is here (I'll add a comment to explain it) is because a title with multiple words with spaces in between is allowed, so this test checks that the space character is included in the regular expression that captures the title. That is the nature of the rest of the tests as well, just check that the regular expression captures all the allowed characters and also captures multiple characters. Like before, the cost of mistakes is minimal since the navigation table is only taken as input on the initial run, and from there is generated and read by the action only (unless a user messes with it in which case it will just end up getting overridden by the action again anyway).

Reply:
Ok if at some point the title is trimmed I get you argument.
If the case you want to test is multiple words separated with a space, why don't you simply use a test with `a b` and expect `a b` as result, that will be self explaining ?

Reply:
Good point, fixed